bait-files/src/test/java/com/google/sps/UsernameTakenTest.java

```diff
@@ -0,0 +1,159 @@
+  }
+
+  @Before 
+  public void setUp() throws InterruptedException, IOException {
+    helper.reset();
+    ObjectifyFactory factory =
+        new ObjectifyFactory(helper.getOptions().getService());
+    ObjectifyService.init(factory);
+    ObjectifyService.register(User.class);
+    objectify = ObjectifyService.begin();
+  }
+
+  @After
+  public void tearDown() throws InterruptedException, IOException, TimeoutException {
+//       try {
+        objectify.close();
+//         helper.stop();
+//       } catch(Exception e) {
+//           System.out.println(e);
+//           System.out.println("TEARDOWN FAILED");
+//       }
+  }
+
+  @AfterClass 
+    public static void oneTimeTearDown() throws InterruptedException, IOException, TimeoutException {
+      helper.stop();
+  }
+
+  @Test
+  public void doGet_whenChosenUsernameInDb_returnsTrue() throws IOException, ServletException {
+    // add a User object
+    ObjectifyService.ofy()
+        .save()
+        .entity(new User("1234321", "Drew", "Andrew", "Seaman"))
+        .now();
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    when(request.getParameter("username")).thenReturn("Drew");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new UsernameTakenServlet().doGet(request, response);
+
+    assertTrue(stringWriter.toString().contains("true"));
+  }
+
+    @Test
+  public void doGet_whenDifferentUsernameInDb_returnsFalse() throws IOException, ServletException {
+    // add a User object
+    ObjectifyService.ofy()
+        .save()
+        .entity(new User("1234321", "Drew", "Andrew", "Seaman"))
+        .now();
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    when(request.getParameter("username")).thenReturn("Chad");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new UsernameTakenServlet().doGet(request, response);
+
+    assertTrue(stringWriter.toString().contains("false"));
+  }
+
+  @Test
+  public void doGet_nothingInDb_returnsFalse() throws IOException, ServletException {
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    when(request.getParameter("username")).thenReturn("Drew");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new UsernameTakenServlet().doGet(request, response);
+
+    assertTrue(stringWriter.toString().contains("false"));
+  }
+
+  @Test
+  public void doGet_usernameIsNull_returnsTrue() throws IOException, ServletException {
```

Comment:
Maybe a comment for *why* this returns true when null? Same w below.

Reply:
Added comment